src/lib/get-api-url.js

```diff
@@ -13,17 +13,13 @@ const getApiUrl = (cmdArgs, env = {}) => {
     return false;
   }
 
-  if (cmdArgs.archive) {
-    return '--archive mode';
-  }
-
   if (cmdArgs.user && !cmdArgs.instance) {
     error('The --user switch can only be used if followed by --instance');
     return false;
   }
 
   let instanceUrl;
-  if (cmdArgs.local) {
+  if (cmdArgs.local || cmdArgs.archive) {
```

Comment:
I'm not 100% comfortable with this change, in part because it'll potentially affect other actions. In the case of form validation it may also validate against the wrong version of Enketo because it'll find whatever API you're running locally even if you didn't specifically request it and validate against that rather than the version you'll be pushing to.

I assume this is to fix the other actions mentioned in [your comment](https://github.com/medic/medic-conf/issues/396#issuecomment-823323674) but I think we should fix those by checking `environment.isArchiveMode` rather than returning an unexpected URL.

Reply:
Yes in archive mode medic-conf does connect with the local instance if it is available, but does not push any report at least so not sure why exactly it does the connection, but now with the new XForm validation the API URL is needed, unless there is another hack I can use.

I checked that the change does not affect the iteration as much I could understand it, eg. I changed the content of a report, then execute medic-conf in archive mode, and the change was not uploaded to the API as previous versions of medic-conf.

I can work in a different patch, but in the meantime @kennsippell could you please tell us whether our understanding is right and why the current implementation in master avoids the URL to be set in archive mode? I see you were one of the developer involved in this feature (https://github.com/medic/medic-conf/commit/5b8f8937fefe1389dfd1f9e6bb7aa6c9c6e1ca2d), that is why I'm asking for your advice please.